clang/test/CodeGenSYCL/sycl-framework-metadata2.cpp

```diff
@@ -0,0 +1,27 @@
+// Test checks that !sycl-framework metadata is emitted only to functions
+// whose topmost namespace is sycl.
+
+// RUN: %clang_cc1 -fsycl-is-device -triple spir64-unknown-unknown -emit-llvm %s -o - | FileCheck %s
+
+// Check that 'anonymous namespace'::sycl::* functions are not marked with
+// !sycl-framework metadata since topmost namespace is anonymous
+// instead of sycl.
+namespace {
```

Comment:
Can't you add this test to sycl-framework-metadata1.cpp. Why is this second file needed?

Reply:
I think because it leads to an ambiguity with ::sycl otherwise.

Reply:
Yes, this is because of ambiguity.

Reply:
Btw, this is also the reason why I can't use a sycl::kernel_single_task mock from `Inputs/sycl.hpp`